x-pack/plugins/aiops/public/embeddables/change_point_chart/embeddable_change_point_chart_factory.tsx

```diff
@@ -0,0 +1,229 @@
+import {
+  apiHasExecutionContext,
+  fetch$,
+  initializeTimeRange,
+  initializeTitles,
+  useBatchedPublishingSubjects,
+} from '@kbn/presentation-publishing';
+import React, { useMemo } from 'react';
+import fastIsEqual from 'fast-deep-equal';
+import useObservable from 'react-use/lib/useObservable';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { getChangePointDetectionComponent } from '../../shared_components';
+import { initializeChangePointControls } from './initialize_change_point_controls';
+import type { AiopsPluginStart, AiopsPluginStartDeps } from '../../types';
+import type {
+  ChangePointEmbeddableApi,
+  ChangePointEmbeddableRuntimeState,
+  ChangePointEmbeddableState,
+} from './types';
+
+export interface EmbeddableChangePointChartStartServices {
+  data: DataPublicPluginStart;
+}
+
+export type EmbeddableChangePointChartType = typeof EMBEDDABLE_CHANGE_POINT_CHART_TYPE;
+
+export const getDependencies = async (
+  getStartServices: StartServicesAccessor<AiopsPluginStartDeps, AiopsPluginStart>
+) => {
+  const [
+    { http, uiSettings, notifications, ...startServices },
+    { lens, data, usageCollection, fieldFormats },
+  ] = await getStartServices();
+
+  return {
+    http,
+    uiSettings,
+    data,
+    notifications,
+    lens,
+    usageCollection,
+    fieldFormats,
+    ...startServices,
+  };
+};
+
+export const getChangePointChartEmbeddableFactory = (
+  getStartServices: StartServicesAccessor<AiopsPluginStartDeps, AiopsPluginStart>
+) => {
+  const factory: ReactEmbeddableFactory<
+    ChangePointEmbeddableState,
+    ChangePointEmbeddableApi,
+    ChangePointEmbeddableRuntimeState
+  > = {
+    type: EMBEDDABLE_CHANGE_POINT_CHART_TYPE,
+    deserializeState: (state) => state.rawState,
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const [coreStart, pluginStart] = await getStartServices();
+
+      const { http, uiSettings, notifications, ...startServices } = coreStart;
+      const { lens, data, usageCollection, fieldFormats } = pluginStart;
+
+      const deps = {
+        http,
+        uiSettings,
+        data,
+        notifications,
+        lens,
+        usageCollection,
+        fieldFormats,
+        ...startServices,
+      };
+
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+
+      const {
+        changePointControlsApi,
+        changePointControlsComparators,
+        serializeChangePointChartState,
+      } = initializeChangePointControls(state);
+
+      const dataLoading = new BehaviorSubject<boolean | undefined>(true);
+      const blockingError = new BehaviorSubject<Error | undefined>(undefined);
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>([
+        await deps.data.dataViews.get(state.dataViewId),
```

Comment:
nit: if you're storing a `dataViewId`, it should really be stored as a `reference`. This is so that if a user exports the Dashboard, the data view used in your change point chart is exported as well. You can take a look at `examples/embeddable_examples/public/react_embeddables/field_list/field_list_react_embeddable.tsx` to see how to accomplish this.

Reply:
Added in https://github.com/elastic/kibana/pull/182246/commits/c19f87b355e7b3c337b926253ef4b53635a6f490